air/src/stack/op_flags/mod.rs

```diff
@@ -0,0 +1,686 @@
+        self.degree6_op_flags[get_op_index(Operation::U32div.op_code())]
+    }
+
+    #[inline(always)]
+    fn u32split(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::U32split.op_code())]
+    }
+
+    #[inline(always)]
+    fn u32assert2(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::U32assert2.op_code())]
+    }
+
+    #[inline(always)]
+    fn u32add3(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::U32add3.op_code())]
+    }
+
+    #[inline(always)]
+    fn u32madd(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::U32madd.op_code())]
+    }
+
+    // ------ Degree 6 non u32 operations  ---------------------------------------------------
+
+    #[inline(always)]
+    fn rpperm(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::RpPerm.op_code())]
+    }
+
+    #[inline(always)]
+    fn mpverify(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::MpVerify.op_code())]
+    }
+
+    #[inline(always)]
+    fn span(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::Span.op_code())]
+    }
+
+    #[inline(always)]
+    fn join(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::Join.op_code())]
+    }
+
+    #[inline(always)]
+    fn split(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::Split.op_code())]
+    }
+
+    #[inline(always)]
+    fn loop_op(&self) -> E {
+        self.degree6_op_flags[get_op_index(Operation::Loop.op_code())]
+    }
+
+    // ------ Degree 4 stack operations  -----------------------------------------------------
+
+    #[inline(always)]
+    fn mrupdate(&self) -> E {
+        self.degree4_op_flags[get_op_index(Operation::MrUpdate(true).op_code())]
+    }
+
+    #[inline(always)]
+    fn push(&self) -> E {
+        self.degree4_op_flags[get_op_index(Operation::Push(ONE).op_code())]
+    }
+
+    #[inline(always)]
+    fn end(&self) -> E {
+        self.degree4_op_flags[get_op_index(Operation::End.op_code())]
+    }
+
+    #[inline(always)]
+    fn repeat(&self) -> E {
+        self.degree4_op_flags[get_op_index(Operation::Repeat.op_code())]
+    }
+
+    #[inline(always)]
+    fn respan(&self) -> E {
+        self.degree4_op_flags[get_op_index(Operation::Respan.op_code())]
+    }
+
+    #[inline(always)]
+    fn halt(&self) -> E {
+        self.degree4_op_flags[get_op_index(Operation::Halt.op_code())]
+    }
+}
+
+trait EvaluationFrameExt<E: FieldElement> {
+    // --- Operation bit accessors ---------------------------------------------------------------
+
+    /// Returns the current value of the specified operation bit in the decoder. It assumes that
+    /// the index is a valid index.
+    fn op_bit(&self, index: usize) -> E;
+
+    /// Returns the value of operation bit extra column which is used to reduce the degree of op
+    /// flags for degree 4 operations where the two most significant bits are ONE.
+    fn helper_register(&self) -> E;
```

Comment:
minor nit: let's rename this to `op_bit_helper`.

Reply:
done